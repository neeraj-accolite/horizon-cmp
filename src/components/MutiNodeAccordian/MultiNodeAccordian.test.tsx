import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MultiNodeAccordion } from './MultiNodeAccordian';
import { MultiNodeAccordionProps } from './MultiNodeAccordian.model';

const items: MultiNodeAccordionProps['items'] = [
  {
    title: 'Accordion Item 1',
    description: 'This is the description for accordion item 1.',
    children: (
      <ul>
        <li>Child Item 1</li>
        <li>Child Item 2</li>
      </ul>
    ),
  },
  {
    title: 'Accordion Item 2',
    description: 'This is the description for accordion item 2.',
    children: (
      <ul>
        <li>Child Item 1</li>
        <li>Child Item 2</li>
      </ul>
    ),
  },
  {
    title: 'Accordion Item 3',
    description: 'This is the description for accordion item 3.',
    children: (
      <ul>
        <li>Child Item 1</li>
        <li>Child Item 2</li>
      </ul>
    ),
  },
];

describe('MultiNodeAccordion', () => {
  test('matches snapshot', () => {
    const { asFragment } = render(<MultiNodeAccordion items={items} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders all accordion items', () => {
    render(<MultiNodeAccordion items={items} />);
    items.forEach((item) => {
      expect(screen.getByText(item.title)).toBeInTheDocument();
    });
  });

  test('toggles accordion content on click', () => {
    render(<MultiNodeAccordion items={items} />);
    items.forEach((item) => {
      const titleElement = screen.getByText(item.title);
      fireEvent.click(titleElement);
      expect(screen.getByText(item.description)).toBeVisible();
      fireEvent.click(titleElement);
    });
  });

  test('applies height and width to each accordion item', () => {
    const height = '200px';
    const width = '300px';
    render(<MultiNodeAccordion items={items} height={height} width={width} />);
    items.forEach((item) => {
      const accordionItem = screen
        .getByText(item.title)
        .closest('.accordionItem');
      expect(accordionItem).toHaveStyle(`height: ${height}`);
      expect(accordionItem).toHaveStyle(`width: ${width}`);
    });
  });
});
