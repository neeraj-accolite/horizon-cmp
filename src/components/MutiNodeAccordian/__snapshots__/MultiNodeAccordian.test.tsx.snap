// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiNodeAccordion matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="accordion"
  >
    <div
      class="accordionItem"
    >
      <div
        class="accordionTitle"
      >
        <span>
          Accordion Item 1
        </span>
        <img
          alt="Toggle Arrow"
          class="icon"
          data-nimg="1"
          decoding="async"
          height="24"
          loading="lazy"
          src="/downArrow.svg"
          style="color: transparent;"
          width="24"
        />
      </div>
      <div
        class="accordionContent "
      >
        This is the description for accordion item 1.
        <div
          class="children"
        >
          <ul>
            <li>
              Child Item 1
            </li>
            <li>
              Child Item 2
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="accordionItem"
    >
      <div
        class="accordionTitle"
      >
        <span>
          Accordion Item 2
        </span>
        <img
          alt="Toggle Arrow"
          class="icon"
          data-nimg="1"
          decoding="async"
          height="24"
          loading="lazy"
          src="/downArrow.svg"
          style="color: transparent;"
          width="24"
        />
      </div>
      <div
        class="accordionContent "
      >
        This is the description for accordion item 2.
        <div
          class="children"
        >
          <ul>
            <li>
              Child Item 1
            </li>
            <li>
              Child Item 2
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="accordionItem"
    >
      <div
        class="accordionTitle"
      >
        <span>
          Accordion Item 3
        </span>
        <img
          alt="Toggle Arrow"
          class="icon"
          data-nimg="1"
          decoding="async"
          height="24"
          loading="lazy"
          src="/downArrow.svg"
          style="color: transparent;"
          width="24"
        />
      </div>
      <div
        class="accordionContent "
      >
        This is the description for accordion item 3.
        <div
          class="children"
        >
          <ul>
            <li>
              Child Item 1
            </li>
            <li>
              Child Item 2
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
